prow/cmd/peribolos/main.go

```diff
@@ -893,7 +903,73 @@ func validateRepos(repos map[string]org.Repo) error {
 	return nil
 }
 
-func configureRepos(client repoClient, orgName string, orgConfig org.Config) error {
+// newRepoUpdateRequest creates a minimal github.RepoUpdateRequest instance
+// needed to update the current repo into the target state.
+func newRepoUpdateRequest(current github.Repo, name string, repo org.Repo) github.RepoUpdateRequest {
+	setString := func(current string, want *string) *string {
+		if want != nil && *want != current {
+			return want
+		}
+		return nil
+	}
+	setBool := func(current bool, want *bool) *bool {
+		if want != nil && *want != current {
+			return want
+		}
+		return nil
+	}
+	repoUpdate := github.RepoUpdateRequest{
+		RepoRequest: github.RepoRequest{
+			Name:             setString(current.Name, &name),
+			Description:      setString(current.Description, repo.Description),
+			Homepage:         setString(current.Homepage, repo.HomePage),
+			Private:          setBool(current.Private, repo.Private),
+			HasIssues:        setBool(current.HasIssues, repo.HasIssues),
+			HasProjects:      setBool(current.HasProjects, repo.HasProjects),
+			HasWiki:          setBool(current.HasWiki, repo.HasWiki),
+			AllowSquashMerge: setBool(current.AllowSquashMerge, repo.AllowSquashMerge),
+			AllowMergeCommit: setBool(current.AllowMergeCommit, repo.AllowMergeCommit),
+			AllowRebaseMerge: setBool(current.AllowRebaseMerge, repo.AllowRebaseMerge),
+		},
+		DefaultBranch: setString(current.DefaultBranch, repo.DefaultBranch),
+		Archived:      setBool(current.Archived, repo.Archived),
+	}
+
+	return repoUpdate
+
+}
+
+func sanitizeRepoDelta(opt options, delta *github.RepoUpdateRequest) []error {
+	var errs []error
+	if delta.Archived != nil && !*delta.Archived {
+		delta.Archived = nil
+		errs = append(errs, fmt.Errorf("asked to unarchive an archived repo, unsupported by GH API"))
+	}
+	if delta.Archived != nil && *delta.Archived && !opt.allowRepoArchival {
+		delta.Archived = nil
+		errs = append(errs, fmt.Errorf("asked to archive a repo but this is not allowed by default (see --allow-repo-archival)"))
+	}
+	if delta.Private != nil && !(*delta.Private || opt.allowRepoPublish) {
+		delta.Private = nil
+		errs = append(errs, fmt.Errorf("asked to publish a private repo but this is not allowed by default (see --allow-repo-publish)"))
+	}
+
+	return errs
+}
+
+func repoExists(repos map[string]github.Repo, name string, previousNames []string) (*github.Repo, bool) {
+	names := []string{name}
+	names = append(names, previousNames...)
+	for _, name := range names {
+		if repo, exists := repos[strings.ToLower(name)]; exists {
```

Comment:
What do we want to do when both `C` and its previous name `P` exist? Should we archive `P`?